packages/plugins/plugin-script/src/capabilities/app-graph-builder.ts

```diff
@@ -8,20 +8,51 @@ import { ScriptType } from '@dxos/functions';
 import { COMPANION_TYPE, SLUG_PATH_SEPARATOR } from '@dxos/plugin-deck/types';
 import { createExtension, type Node } from '@dxos/plugin-graph';
 
-import { SCRIPT_PLUGIN } from '../meta';
+import { meta } from '../meta';
 
 export default (context: PluginsContext) =>
   contributes(Capabilities.AppGraphBuilder, [
+    // TOOD(burdon): Factor out: make generic?
```

Comment:
can you be more specific? do you mean a helper for defining companion nodes?

that may be a useful interim step but imo the real solution is having nodes have effect schema attached to the `type`

Reply:
yeah i meant all/most plugins will need to define a settings panel -- and it should cover the base properties, which are currently contributed to the complementary panel via the space.